Resources/public/js/views/services/plugins/ez-userdraftsplugin.js

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-userdraftsplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the user load plugin
+     *
+     * @module ez-userloadplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Object user drafts plugin.
+     *
+     * In order to use it you need to fire the `loadUserDrafts` event with two parameters:
+     *  - `attributeName` where to store the result
+     *  - `limit` limit of results
+     *
+     * @namespace eZ.Plugin
+     * @class UserDrafts
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.UserDrafts = Y.Base.create('userDraftsPlugin', Y.eZ.Plugin.ViewServiceBase, [], {
+        initializer: function () {
+            this._resetTempVariables();
+            this.onHostEvent('*:loadUserDrafts', this._loadUserDrafts);
+        },
+
+        /**
+         * Loads user drafts. Once this is done, it sets the content in
+         * the attribute defined in the event parameter of the event target.
+         * @protected
+         * @method _loadUser
+         * @param {Object} e loadUser event facade
+         */
+        _loadUserDrafts: function (event) {
+            var host = this.get('host');
+
+            this._target = event.target;
+            this._targetAttributeName = event.attributeName;
+            this._targetLimit = event.limit;
+
+            host.get('app').get('user').loadDrafts(
+                {api: host.get('capi')},
+                Y.bind(this._collectDraftsData, this)
+            );
+        },
+
+        /**
+         * Resets temporary variables
+         *
+         * @protected
+         * @method _resetTempVariables
+         */
+        _resetTempVariables: function () {
+            /**
+             * Holds a temporary reference to `loadUserDrafts` event target
+             *
+             * @property _target
+             * @type {Y.View}
+             * @protected
+             */
+            this._target = null;
```

Comment:
you should not store a state like that in the object because you are forced to reset the state in several places and that makes the whole code flow is bit cumbersome. Instead, you should pass those values as parameters when needed. For instance the attribute name is only needed at the very end of the process, the limit is only needed at the very beginning.